grep-0018-modernize-c++.md

```diff
@@ -0,0 +1,417 @@
+# GREP 0018 -- Modernize C++
+
+- Original Author: Thomas Habets <thomas@habets.se>
+- Champion: Thomas Habets <thomas@habets.se>
+- Status: Draft
+
+History:
+- 20-Jan-2020: Initial Draft
+
+## Abstract
+
+As of early 2020 there is lots of pre-C++11 code in GNU Radio, and
+some other pattern of non-idiomatic C++.
+
+This GREP lays out how to get from here to there with minimal
+breaking.
+
+## Copyright / License
+
+This GREP is licensed as CC-BY-ND. Copyright 2020 Google Inc.
+
+## Motivation
+
+Cleaning up whole patterns of the code 
+
+The legacy patterns this GREP lays out a plan for making the code base:
+
+* more readable: standard components, less code, and explicit const
+* safer: less manual memory management, prevent inherently broken copy
+* faster: fewer pointer indirections, more information to the compiler
+
+## Description
+
+### Open questions
+
+* What level of ABI guarantee does GNU Radio make between and within
+  minor releases?
+
+### More const
```

Comment:
Can you please elaborate these headers a little: "Replace more variables with const where appropriate"

Reply:
On const: That's not really C++11, that's always been good practice. I wonder if it's even worth putting that in this PR. It's also not a *huge* win when we retroactively add const into existing code. Maybe we should focus on bigger wins instead. Usage of const is something that should be flagged during code reviews.

Reply:
I'm adding a middle ground.